pandas/tests/io/excel/test_openpyxl.py

```diff
@@ -116,3 +116,12 @@ def test_to_excel_with_openpyxl_engine(ext):
         ).highlight_max()
 
         styled.to_excel(filename, engine="openpyxl")
+
+
+def test_read_workbook(datapath, ext):
+    # GH 39528
+    filename = datapath("io", "data", "excel", "test1" + ext)
+    wb = openpyxl.load_workbook(filename)
+    result = pd.read_excel(wb, engine="openpyxl")
+    expected = pd.read_excel(filename)
```

Comment:
It cannot hurt to call `wb.close()` before `tm.assert_frame_equal`

Reply:
Thanks @twoertwein, added.